root/static/styles/layout.less

```diff
@@ -126,8 +125,167 @@ pre code {
 /*
 ** Header
 */
-#header {
-    width: 100%;
+.header {
+    @menu-item-text-padding: 8px;
+    @logo-width: 166px;
+    @logo-height: 26px;
+
+    height: 62px;
+    margin-top: 10px;
+    margin-left: 16px;
+    margin-right: 16px;
+    clear: both;
+
+    img.logo {
+        width: @logo-width;
+        height: @logo-height;
+        margin-top: 16px;
+    }
+
+    .right {
+        width: 85%; // fallback for browsers without support for calc()
+        width: ~"calc(100% - @{logo-width} - 26px)";
+        float: right;
+
+        .top {
+            height: 31px;
+            border-bottom: @musicbrainz-purple 2px solid;
+            clear: both;
+            .links-container {
+                float: left;
+                ul.menu {
+                    li {
+                        img.gravatar {
+                            vertical-align: text-top;
+                        }
+                    }
+                }
+            }
+            .search-container {
+                @item-height: 25px;
+                @outer-border-color: fade(@musicbrainz-purple, 50%);
+                @inner-border-color: @medium-border;
+
+                float: right;
+                form {
+                    width: 302px;
+                    margin-top: 3px;
```

Comment:
I would change the `margin-top` to `-2px` to align the search form vertically in the whitespace between the top of the page and the menu bar. Looks more balanced IMO.